qdev_wrappers/dataset/doNd.py

```diff
@@ -0,0 +1,278 @@
+        before_inner_actions: Actions executed before each run of the inner loop
+        after_inner_actions: Actions executed after each run of the inner loop
+        do_plot: should png and pdf versions of the images be saved after the
+            run.
+
+    Returns:
+        The run_id of the DataSet created
+    """
+
+    meas = Measurement()
+    meas.register_parameter(param_set1)
+    param_set1.post_delay = delay1
+    meas.register_parameter(param_set2)
+    param_set1.post_delay = delay2
+
+    for action in enter_actions:
+        # this omits the possibility of passing
+        # argument to enter and exit actions.
+        # Do we want that?
+        meas.add_before_run(action, ())
+
+    for action in exit_actions:
+        meas.add_after_run(action, ())
+
+    for parameter in param_meas:
+        if isinstance(parameter, _BaseParameter):
+            meas.register_parameter(parameter,
+                                    setpoints=(param_set1, param_set2))
+    with meas.run() as datasaver:
+
+        for set_point1 in np.linspace(start1, stop1, num_points1):
+            param_set1.set(set_point1)
+            for action in before_inner_actions:
+                action()
+            for set_point2 in np.linspace(start2, stop2, num_points2):
+                param_set2.set(set_point2)
+                output = []
+                for i, parameter in enumerate(param_meas):
+                    if isinstance(parameter, _BaseParameter):
+                        output.append((parameter, parameter.get()))
+                    elif callable(parameter):
+                        parameter()
+                datasaver.add_result((param_set1, set_point1),
+                                     (param_set2, set_point2),
+                                     *output)
+            for action in after_inner_actions:
+                action()
+
+        dataid = datasaver.run_id
+
+    if do_plot is True:
+        ax, cbs = _save_image(datasaver)
+    else:
+        ax = None,
+        cbs = None
+
+    return dataid, ax, cbs
+
+
+def _save_image(datasaver) -> AxesTupleList:
+    """
+    Save the plots created by datasaver as pdf and png
+
+    Args:
+        datasaver: a measurement datasaver that contains a dataset to be saved
+            as plot.
+
+    """
+    plt.ioff()
+    dataid = datasaver.run_id
+    start = time.time()
+    axes, cbs = plot_by_id(dataid)
+    stop = time.time()
+    print(f"plot by id took {stop-start}")
+
+    mainfolder = config.user.mainfolder
+    experiment_name = datasaver._dataset.exp_name
+    sample_name = datasaver._dataset.sample_name
+
+    storage_dir = os.path.join(mainfolder, experiment_name, sample_name)
+    os.makedirs(storage_dir, exist_ok=True)
+
+    png_dir = os.path.join(storage_dir, 'png')
+    pdf_dif = os.path.join(storage_dir, 'pdf')
+
+    os.makedirs(png_dir, exist_ok=True)
+    os.makedirs(pdf_dif, exist_ok=True)
+
+    save_pdf = True
+    save_png = True
```

Comment:
I agree that it is a good idea, to just have these sensible defaults for now instead of loading it from the config